rts/System/TraceMemory.cpp

```diff
@@ -0,0 +1,22 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+#if defined(TRACY_ENABLE)
+
+#include <new>
+#include <cstdlib>
+#include <tracy/Tracy.hpp>
+
+void* operator new(std::size_t count)
```

Comment:
If operator new/delete override indeed works perhaps chain that call with something like `spring::operator new(size)`, which would be defined elsewhere and cover possible integration of https://github.com/microsoft/mimalloc

Reply:
The overrides I'm providing are very basic, https://github.com/microsoft/mimalloc/blob/master/include/mimalloc-new-delete.h provides all the different variants, and that's still a performance optimization, as to integrate that properly you should override malloc and others during linking first.

I'm also not 100% sure I'm covering all the cases anyway, so, I think that adding any calls like that is a bit premature, as I'm not confident we know how the final shape of the code will look like: the current code is extremely minimal.